link_linux.go

```diff
@@ -1580,9 +1586,57 @@ func LinkDeserialize(hdr *unix.NlMsghdr, m []byte) (Link, error) {
 	}
 	*link.Attrs() = base
 
+	// If the tuntap attributes are not updated by netlink due to
+	// an older driver, use sysfs
+	if link != nil && linkType == "tun" {
+		tuntap := link.(*Tuntap)
+
+		if tuntap.Mode == 0 {
+			ifname := tuntap.Attrs().Name
+			if flags, err := readSysPropAsInt64(ifname, "tun_flags"); err == nil {
+
+				if (flags & unix.IFF_TUN) != 0 {
+					tuntap.Mode = unix.IFF_TUN
+				} else if (flags & unix.IFF_TAP) != 0 {
+					tuntap.Mode = unix.IFF_TAP
+				}
+
+				if flags&unix.IFF_PERSIST == 0 {
+					tuntap.NonPersist = true
+				} else {
+					tuntap.NonPersist = false
+				}
+
+			}
+			if owner, err := readSysPropAsInt64(ifname, "owner"); err == nil && owner > 0 {
+				tuntap.Owner = uint32(owner)
+			}
+			if group, err := readSysPropAsInt64(ifname, "group"); err == nil && group > 0 {
+				tuntap.Group = uint32(group)
+			}
+		}
+	}
+
 	return link, nil
 }
 
+func readSysPropAsInt64(ifname, prop string) (int64, error) {
+	var num int64
+
+	fname := fmt.Sprintf("/sys/class/net/%s/%s", ifname, prop)
+
+	contents, err := ioutil.ReadFile(fname)
+	if err != nil {
+		return num, err
```

Comment:
on same line to keep code lean, after taking care of previous comment, it is perfectly fine to`return 0, err` here.